src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -756,3 +757,30 @@ func (c *controller) UploadLogs(ctx context.Context, cancellog context.CancelFun
 		}
 	}
 }
+
+func (c controller) SetReadyState() {
+	c.log.Infof("Start waiting to be ready")
+	_ = utils.WaitForPredicate(WaitTimeout, 1*time.Second, func() bool {
+		_, err := c.ic.GetCluster(context.TODO())
+		if err != nil {
+			c.log.WithError(err).Warningf("Failed to connect to assisted service")
+			return false
+		}
+
+		_, err = c.kc.ListNodes()
+		if err != nil {
+			c.log.WithError(err).Warningf("Failed to connect to ocp cluster")
+			return false
+		}
+
+		c.log.Infof("I am ready, spawning event")
```

Comment:
can it be more descriptive message? Think about customer/support. What does it mean ready?